utils.py

```
@@ -1,4 +1,49 @@
+    
+    Returns:
+        Git commit hash as a string, or empty string if not available.
+    """
+    try:
+        if repo_path:
+            # Change to repo directory for git command
+            original_dir = os.getcwd()
+            try:
+                os.chdir(repo_path)
+                result = subprocess.run(
+                    ['git', 'rev-parse', 'HEAD'],
```

Ruff (0.14.4)

Starting a process with a partial executable path

(S607)